meta-mbl-distro/classes/mbl-partitions.bbclass

```diff
@@ -24,138 +24,407 @@ inherit mbl-artifact-names
 # will be used in your case.
 MBL_FLASH_ERASE_BLOCK_SIZE_KiB ?= "16384"
 
+MBL_DEFAULT_ALIGN_KiB ?= "${MBL_FLASH_ERASE_BLOCK_SIZE_KiB}"
+MBL_DEFAULT_FSTYPE ?= "ext4"
+MBL_DEFAULT_MOUNT_OPTS ?= "defaults"
+MBL_DEFAULT_FS_FREQ ?= "0"
+MBL_DEFAULT_FS_PASSNO ?= "0"
 
+# ------------------------------------------------------------------------------
+# Partition variables
+# ------------------------------------------------------------------------------
 # We have two different flavours of "partition" in MBL:
 # * Normal file system partitions, recorded in the partition table.
 # * Raw non-file system "partitions", not recorded in the partition table.
+# 
+# For each of MBL's "partitions" there are the following variables:
+# * MBL_<partition>_ALIGN_KiB   - minimum alignment in KiB
+# * MBL_<partition>_SIZE_KiB    - size in KiB
+# * MBL_<partition>_SIZE_MiB    - size in MiB
+# * MBL_<partition>_OFFSET1_KiB - offset of first bank in KiB
+# * MBL_SKIP_<partition>        - skip this partition?
+# * MBL_<partition>_NO_FS       - is this a raw partition (not a file system partition)?
+# * MBL_<partition>_IS_BANKED   - are there two banks of this partition
+#
+# For partitions with two banks there is:
+# * MBL_<partition>_OFFSET2_KiB - offset of second bank in KiB
+#
+# For non-file system partitions there is:
+# * MBL_<partition>_FILENAME    - file in image deploy dir
+#
+# For file system partitions there is:
+# * MBL_<partition>_MOUNT_POINT - where this partition is mounted in the Linux file system)
+# * MBL_<partition>_LABEL       - label of partition
+# * MBL_<partition>_FSTYPE      - type of file system
+# * MBL_<partition>_MOUNT_OPTS  - mount options
+# * MBL_<partition>_FS_FREQ     - fifth fstab(5) field
+# * MBL_<partition>_FS_PASSNO   - sixth fstab(5) field
+#
+# Notes:
+# * MBL_<partition>_ALIGN_KiB may be left unset outside of this file and a
+#   default value will be used. For WKS_BOOTLOADER1 the default value is "1".
+#   For other partitions the default value is ${MBL_FLASH_ERASE_BLOCK_SIZE_KiB}.
+#
+# * If both MBL_<partition>_SIZE_KiB and MBL_<partition>_SIZE_MiB are set
+#   then they must not contradict each other.
+#
+# * If both MBL_<partition>_SIZE_KiB and MBL_<partition>_SIZE_MiB are left
+#   unset outside of this file then default sizes will be used.
+#
+# * For file system partitions, the size must be a multiple of 1MiB.
+#
+# * If WKS_BOOTLOADER1 is not skipped then WKS_BOOTLOADER1_OFFSET1_KiB
+#   must be set outside of this file - there is no default value because this
+#   is always board specific. For WKS_BOOTLOADER1 and WKS_BOOTLOADER_FS,
```

Comment:
Did you mean `WKS_BOOTLOADER2`?

Reply:
Yes. https://github.com/ARMmbed/meta-mbl/pull/617/commits/69b0c1c8779d30746ad02fb530710d8b60023428.